index.js

```diff
@@ -405,34 +321,96 @@ function loadPageState() {
     bookAuthorFilter.value = filterState.bookAuthor;
     yearFilter.value = filterState.year;
-    currentPage = filterState.currentPage;
   }
-}
+};
 
-document.addEventListener('DOMContentLoaded', () => {
-  loadPageState(); //Calls the loadPageState function when the page loads fully to checks if there is any saved state
-  // getBooksData();
-  getBooksDataPromises();
-});
-
-// searchBtn.addEventListener('click', getBooksData);
-searchBtn.addEventListener('click', getBooksDataPromises);
-
-document.querySelectorAll('.table-sortable th').forEach((headerCell) => {
-  headerCell.addEventListener('click', () => {
-    console.log("headerCell.parentElement.children");
-    console.log(headerCell.parentElement.children);
-    console.log("headerCell");
-    console.log(headerCell);
-    const tableElement = headerCell.closest('table');
-    const headerIndex = Array.prototype.indexOf.call(
-      headerCell.parentElement.children,
-      headerCell
-    );
-    const currentIsAsc = headerCell.classList.contains('th-sort-asc');
-    // Only sort if the column is Price (index 2) or Publication Year (index 4)
-    if (headerIndex == 3 || headerIndex == 5) {
-      sortTableCol(tableElement, headerIndex, !currentIsAsc);
+const handleSort = (valType, tableType) => {
+  console.log(valType);
+  console.log(tableType);
+  if (tableType === 'similar-books-table') {
+    console.log('In If of Similar  Books');
+    console.log(finalSimilarBookList);
+    similarTableBody.innerHTML = '';
+    if (valType === 'price') {
+      finalSimilarBookList.sort((a, b) => (isAsc ? a.price - b.price : b.price - a.price));
+    } else if (valType === 'year') {
+      finalSimilarBookList.sort((a, b) => (
+        isAsc ? a.publicationYear - b.publicationYear : b.publicationYear - a.publicationYear
+      ));
     }
-  });
-});
+    isAsc = !isAsc;
+    for (let i = 0; i < finalSimilarBookList.length; i += 1) {
+      const coverImage = finalSimilarBookList[i].coverImage || defaultImage;
+      const rowEntry = document.createElement('tr');
+      rowEntry.innerHTML = `
+        <td><img src="${coverImage}" alt="${finalSimilarBookList[i].bookName}" style="width:50px;height:50px;object-fit:contain"></td>
+        <td>${finalSimilarBookList[i].bookName}</td>
+        <td>${finalSimilarBookList[i].genre}</td>
+        <td>$ ${finalSimilarBookList[i].price}</td>
+        <td>${finalSimilarBookList[i].author}</td>
+        <td>${finalSimilarBookList[i].publicationYear}</td>`;
+      similarTableBody.appendChild(rowEntry);
+    }
+  } else if (tableType === 'all-books-table') {
+    allBooksTableBody.innerHTML = '';
+    if (valType === 'price') {
+      data.sort((a, b) => (isAsc ? a.price - b.price : b.price - a.price));
+    } else if (valType === 'year') {
+      data.sort((a, b) => (isAsc ? a.publicationYear - b.publicationYear : b.publicationYear - a.publicationYear));
+    }
+    isAsc = !isAsc;
+    renderAllBooks();
+  }
+};
+
+const handleSearch = () => {
```

Comment:
This code for **handleSearch** and **filterData** looks unoptimized and non-readable, can you try optimizing like below and test.
```
const filterData = (val, filterType, currentData) => {
  const lowerCaseVal = val.toLowerCase();
  switch (filterType) {
    case 'bookName':
      return currentData.filter((book) => book.bookName.toLowerCase().includes(lowerCaseVal));
    case 'genre':
      return currentData.filter((book) => book.genre.toLowerCase().includes(lowerCaseVal));
    case 'priceMin':
      return currentData.filter((book) => book.price >= val);
    case 'priceMax':
      return currentData.filter((book) => book.price <= val);
    case 'author':
      return currentData.filter((book) => book.author.toLowerCase().includes(lowerCaseVal));
    case 'publicationYear':
      return currentData.filter((book) => book.publicationYear.toString() === val);
    default:
      return currentData;
  }
};

const handleSearch = () => {
  let filteredData = data;
  const filters = [
    { value: bookNameFilter.value, type: 'bookName' },
    { value: bookGenreFilter.value, type: 'genre' },
    { value: priceMinFilter.value, type: 'priceMin' },
    { value: priceMaxFilter.value, type: 'priceMax' },
    { value: bookAuthorFilter.value, type: 'author' },
    { value: yearFilter.value, type: 'publicationYear' }
  ];

  filters.forEach(filter => {
    if (filter.value) {
      filteredData = filterData(filter.value, filter.type, filteredData);
    }
  });

  if (filters.some(filter => filter.value)) {
    filteredData.sort((a, b) => a.price - b.price);
    similarBookContainer.style.display = 'block';
    renderSearchedBookDetails(filteredData);
  } else {
    renderSearchedBookDetails([]);
  }

  savePageState();
};
```

Reply:
Done